documentation/DEPLOYMENT-AZD-TERRAFORM.md

```diff
@@ -0,0 +1,139 @@
+
+### The "variables.tf" file
+
+This files defines the input variables for the `main.tf` file. They can be seen like function arguments and are used in the setup of the infrastructure to e.g., provide the value for a specific resource configuration.
+
+Here we also placed our quickstarter specific variables that define the access to SAP system's OData services.
+
+### The "output.tf" file
+
+This file defines the output of the provisioning e.g., the URL endpoint of the application. The values correspond to function return values. They will be displayed in the CLI output when the provisioning was successfully executed.
+
+### The "main.tfvars.json" file
+
+This files is the so called **variable definitions**  file. This is one way to set the input variables defined in the `variables.tf`.
+
+### The "main.tf" file
+
+This file is the core of the Terraform infrastructure configuration. Here we define the infrastructure setup of our application by combining the different building blocks provided via the Terraform modules. One example is the configuration of the module to provision the Azure App Service:
+
+```tf
+module "api" {
+  source         = "./modules/appservicenode"
+  location       = var.location
+  rg_name        = azurerm_resource_group.rg.name
+  resource_token = local.resource_token
+
+  tags               = merge(local.tags, { "azd-service-name" : "sap-cloud-sdk-api" })
+  service_name       = "sap-cloud-sdk-api"
+  appservice_plan_id = module.appserviceplan.APPSERVICE_PLAN_ID
+  app_settings = {
+    "SCM_DO_BUILD_DURING_DEPLOYMENT"        = "true"
+    "AZURE_KEY_VAULT_ENDPOINT"              = module.keyvault.AZURE_KEY_VAULT_ENDPOINT
+    "APPLICATIONINSIGHTS_CONNECTION_STRING" = module.applicationinsights.APPLICATIONINSIGHTS_CONNECTION_STRING
+    "ODATA_URL"                             = var.oDataUrl
+    "ODATA_USERNAME"                        = var.oDataUsername
+    "ODATA_USERPWD"                         = "@Microsoft.KeyVault(SecretUri=${module.keyvault.AZURE_KEY_VAULT_ENDPOINT}secrets/${local.abbrKeyVaultVaults}secret-odata-password)"
+    "APIKEY"                                = "@Microsoft.KeyVault(SecretUri=${module.keyvault.AZURE_KEY_VAULT_ENDPOINT}secrets/${local.abbrKeyVaultVaults}secret-apikey)"
+    "APIKEY_HEADERNAME"                     = var.ApiKeyHeaderName
+  }
+
+  app_command_line = ""
+
+  always_on = false
+  identity = [{
+    type = "SystemAssigned"
+  }]
+}
+```
+
+As you can see we use the variables defined in the `variables.tf` file to configure e.g., the app settings. The components are basically the same as in the Bicep flow, so we do not repeat them in detail here.
+
+## Mind the differences
+
+The Terraform-based setup in the `azd` was introduced a later than the òne using Bicep. Due to that and due to the differences between them, some things work differently. Here are two things that we think are worth to mention.
```

Comment:
typo the "one"

Reply:
Fixed with commit <https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart/commit/c59ed5bcf790245e421476d477e5e3ebf1566450>